src/libraries/Reserves.sol

```diff
@@ -0,0 +1,40 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.20;
+
+import {Currency} from "../types/Currency.sol";
+
+library Reserves {
+    /// uint256(keccak256("ReservesOf")) - 1
+    uint256 constant RESERVES_OF_SLOT = uint256(0x1e0745a7db1623981f0b2a5d4232364c00787266eb75ad546f190e6cebe9bd95);
+    /// @notice The transient reserves for pools with no balance is set to the max as a sentinel to track that it has been synced.
+    uint256 public constant ZERO_BALANCE = type(uint256).max;
+
+    /// @notice Thrown when the transient reserves are not in sync with the current balance
```

Comment:
nit: this comment isnt quite right. When `ReservesMustBeSynced` is emitted, someone is calling `settle` which means that the current balance != transient reserves because they want to pay the difference.
```
/// @notice Thrown when someone has not called sync before calling settle
```
or something like that